scripts/tuple_tree_generator/tuple_tree_generator/templates/struct.h.tpl

```diff
@@ -142,9 +155,76 @@ public:
   bool operator>(const /*= struct.name =*/ &Other) const { return key() > Other.key(); }
 
   /** else **/
-  bool operator==(const /*= struct.name =*/ &Other) const = default;
+  bool operator==(const /*= struct.name =*/ &Other) const {
+    /**- for field in struct.fields **/
+    if (/*= field.name =*/() != Other./*= field.name =*/())
+      return false;
+    /**- endfor **/
+    return true;
+  }
   /** endif **/
 
+  /**- if emit_tracking **/
+  public:
+  template <size_t I> bool wasNthFieldAccessed() const {
+    if constexpr (false)
+      return false;
+    /**- for field in struct.all_fields **/
+    else if constexpr (I == /*= loop.index0 =*/)
+        return /*= field.name =*/Tracker.isSet();
+    /**- endfor -**/
+    else
+      revng_assert(false);
+    return false;
+  }
+
+  template <size_t I> void popTracker() const {
+    if constexpr (false)
+      ;
+    /**- for field in struct.all_fields **/
+    else if constexpr (I == /*= loop.index0 =*/)
+        /*= field.name =*/Tracker.pop();
+    /**- endfor -**/
+    else
+      revng_assert(false);
+  }
+
+  void popKeysTrackers() const {
+    /**- for field in struct.key_fields **/
+    /*= field.name =*/Tracker.pop();
+    /**- endfor -**/
+  }
+
+  void pushKeysTrackers() const {
+    /**- for field in struct.key_fields **/
+    /*= field.name =*/Tracker.push();
+    /**- endfor -**/
+  }
+
+  template <size_t I> void pushTracker() const {
+    if constexpr (false)
+      ;
+    /**- for field in struct.all_fields **/
+    else if constexpr (I == /*= loop.index0 =*/)
+        /*= field.name =*/Tracker.push();
+    /**- endfor -**/
+    else
+      revng_assert(false);
+  }
+
+  template<size_t I> void clearTracker() const {
```

Comment:
Let's be consistent with terminology. Elsewhere we are using `reset`. Let's stick with `clear` everywhere.